src/modules/account/account.service.ts

```diff
@@ -0,0 +1,159 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { CreateAccountDto } from './dtos/create-account.dto';
+import { generate } from 'generate-password';
+import { UserRoles } from '@prisma/client';
+import { PrismaService } from '../prisma/prisma.service';
+import { UpdateAccountDto } from './dtos/update-account.dto';
+import { ExceptionsHandler } from '@nestjs/core/exceptions/exceptions-handler';
+
+@Injectable()
+export class AccountService {
+  constructor(private prisma: PrismaService) {}
+
+  async createAccount(data: CreateAccountDto, userId: number) {
+    try {
+      if (!userId) return null;
+
+      if (Object.keys(data).length === 0) {
+        throw new BadRequestException('No data provided.');
```

Comment:
@Akeron44 Input/Request validation could be moved to a separate file just for clarity